application/src/main/java/uk/nhs/hee/tis/revalidation/validator/TraineeRecommendationRecordDTOValidator.java

```diff
@@ -45,32 +57,56 @@ public void validate(Object target, Errors errors) {
 
     if (errors.getErrorCount() == 0) {
       final var recordDTO = (TraineeRecommendationRecordDto) target;
+
       if (!StringUtils.hasLength(recordDTO.getGmcNumber())) {
         errors.reject("GmcNumber", "Gmc Number can't be empty or null");
+        return;
       }
+
       if (!StringUtils.hasLength(recordDTO.getRecommendationType())) {
         errors.reject("RecommendationType", "Recommendation type can't be empty or null");
       } else {
-        final var recommendationType = RecommendationType
-            .valueOf(recordDTO.getRecommendationType());
+        final var recommendationType = RecommendationType.valueOf(
+            recordDTO.getRecommendationType());
+
         if (RecommendationType.DEFER.equals(recommendationType)) {
-          if (recordDTO.getDeferralDate() == null || recordDTO.getDeferralDate().isBefore(now())) {
-            errors.reject("DeferralDate", "Deferral date can't be empty or in past");
-          }
-          //Doctor X has a submission due date of 120 days from today (today <= 120 days)
-          if (recordDTO.getGmcSubmissionDate() == null
-              || (ChronoUnit.DAYS.between(now(), recordDTO.getGmcSubmissionDate())) > 120) {
-            errors.reject("GmcSubmissionDate",
-                "Deferral is not permitted at this time since submission due date is greater than 120 days from today");
-          }
-          if (!StringUtils.hasLength(recordDTO.getDeferralReason())) {
-            errors.reject("DeferralReason", "Deferral Reason can't be empty or null");
-          } else if (recordDTO.getDeferralReason().equalsIgnoreCase(INSUFFICIENT_EVIDENCE)
-              && !StringUtils.hasLength(recordDTO.getDeferralSubReason())) {
-            errors.reject("DeferralSubReason", "Deferral Sub Reason can't be empty or null");
-          }
+          validateDefer(recordDTO, errors);
         }
       }
     }
   }
+
+  protected void validateDefer(TraineeRecommendationRecordDto recordDto, Errors errors) {
+
+    if (recordDto.getDeferralDate() == null || recordDto.getDeferralDate().isBefore(now())) {
+      errors.reject("DeferralDate", "Deferral date can't be empty or in past");
+    } else {
+      validateIfDeferAllowed(recordDto, errors);
+    }
```

Comment:
It's a bit late but also looks like a change from "validate both" to "validate at least one".  I can't see that this is a conscious choice / why would we provide less information to the client & make them resubmit to get the next Error?